Add link underline offset and make underline thicker on hover

frankieroberto opened the pull request:
## Description

This PR updates links (and link-like elements such as details) so that the underline consistently thinner and offset further away from the link text, improving readability.

It also introduces new mixins and modifier classes for:

- text colour links
- muted colour links
- links without underlines

| Before | After |
|----|-----|
| <img width="636" height="912" alt="Links before" src="https://github.com/user-attachments/assets/dbf479e1-6554-461d-8302-dad8b7f8f9a8" />  | <img width="636" height="912" alt="Links after" src="https://github.com/user-attachments/assets/fd17ebf4-b1d9-4829-ab0f-31da49560d78" /> |

## Checklist

- [ ] Tested against our [testing policy](https://github.com/nhsuk/nhsuk-frontend/blob/master/docs/contributing/testing.md) (Resolution, Browser & Accessibility)
- [x] Follows our [coding standards and style guide](https://github.com/nhsuk/nhsuk-frontend/blob/master/docs/contributing/coding-standards.md)
- [x] CHANGELOG entry

frankieroberto commented:
> Are the before/after images for the first row the correct way round?

Yep - the only difference in the first row is a 1px offset for the underline.

frankieroberto commented:
> Oh. Is that enough differentiation? Should the underline not get thicker? Should it not be consistent with the other link example shown? (On my phone, so can't test right now, so apologies for perhaps silly questions!)

Check when you’re back at a laptop! It’s offset by 1px, and gets thicker (from 1px to 3px) on hover.

Basically, the exact same as GOVUK (whose code I copied).

paulrobertlloyd commented:
- Did you mean to have navigation links in the header underlined? Are there other places this adds underlines where we don’t currently have them?
- It looks although on GOV.UK, the hover state doesn’t add any text decoration gaps between descenders, which makes for a less distracting transition. We should probably do the same.

frankieroberto commented:
> Did you mean to have navigation links in the header underlined? Are there other places this adds underlines where we don’t currently have them?

I hadn't quite realised how widely (although inconsistently) the link mixin was used. I’ve now added a `$link-underlines` boolean option to all the link mixins, so that we can explicitly specify which components should use underlines for the links (or link-like components) and which ones shouldn't.

For now, I've added/kept underlines to Back links, Breadcrumbs, Cards, Contents list, Details, Error summary, Expander, Footer, Hero, Summary list, Pagination

There's no underlines on: Action link, Header

All up for debate though!

> It looks as though on GOV.UK the hover state doesn’t add any text decoration gaps between descenders, which makes for a less distracting transition. We should probably do the same.

Fixed in 15a8d0d35a179a20db7bc593528c174be37de22b.

anandamaryon1 commented on packages/nhsuk-frontend/src/nhsuk/core/tools/_links.scss:
```diff
@@ -79,16 +91,29 @@
+
+      // Disable ink skipping on underlines on hover. Browsers haven't
+      // standardised on this part of the spec yet, so set both properties
+      text-decoration-skip-ink: none; // Chromium, Firefox
```
Maybe increasing it slightly, with ink-skip, is better actually…

As-is
![image](https://github.com/user-attachments/assets/77eb969d-730b-449f-98a0-906b8e1b36de)

Increased
![image](https://github.com/user-attachments/assets/2629ce39-e146-486b-b0b8-4527ed390648)

frankieroberto replied:
@anandamaryon1 in the commit above it’s only disabling ink skipping on hover - as with the thicker underline on hover, browsers seem to increase amount of ink skipping, which makes for a bit of a jarring transition. With the thicker underline, ink skipping seems less useful?

Whether we should have ink skipping or not in the regular state, and how much the offset should be, I’m not sure!

frankieroberto commented on packages/nhsuk-frontend/src/nhsuk/components/details/_index.scss:
```diff
@@ -25,15 +25,11 @@
-  &:hover {
-    color: $nhsuk-link-hover-color;
-  }
+  @include nhsuk-link-style($link-underlines: true);
```
Err, could do, I hadn’t spotted that.

Our current `nhsuk-link-style` mixin is set up in quite a different way though, and already has 4 keyword arguments, so this seemed consistent with that. We also need the option to use the `nhsuk-link-style-white` mixin with no underlines.

I found the way the link mixins were structured (and how they include each other) all quite confusing too (and govuk's is confusing too, but in a different way), so open to better ideas!

paulrobertlloyd commented:
Have rebased this with `main`, and removed commits relating to the header, as we’ll be dealing with that separately, at least for the next release.

anandamaryon1 commented:
Reviewed at Design System Review Panel. Generally happy with it, but we just need to iron out how it applies to the header and other components, such as the Action link, to ensure we have a clear rationale to what is underlined and what is not. (currently in this PR everything is underlined, which may be best).

I also notice the expander isn't showing a hover colour. 

In chrome (but not safari) I get this slightly ugly rendering of `g` and `y` descenders. (less so on high-res / retina screen).

Chrome
<img width="103" height="36" alt="image" src="https://github.com/user-attachments/assets/5b5ce3f2-44d2-4b7b-9617-cb7e42195974" />

Safari
<img width="102" height="39" alt="image" src="https://github.com/user-attachments/assets/d1cc5fe7-7cd1-4df7-91ab-e6eb3590f22a" />

I'll do some more detailed review and explore some options.
